gui-daemon/xside.c

```
@@ -454,28 +528,29 @@ static void mkghandles(Ghandles * g)
     g->context = XCreateGC(g->display, g->root_win, 0, NULL);
-    g->wmDeleteMessage =
-        XInternAtom(g->display, "WM_DELETE_WINDOW", True);
+    g->wmDeleteMessage = intern_atom(g, "WM_DELETE_WINDOW", True);
     g->clipboard_requested = 0;
     g->clipboard_xevent_time = 0;
     snprintf(tray_sel_atom_name, sizeof(tray_sel_atom_name),
          "_NET_SYSTEM_TRAY_S%u", DefaultScreen(g->display));
-    g->tray_selection =
-        XInternAtom(g->display, tray_sel_atom_name, False);
-    g->tray_opcode =
-        XInternAtom(g->display, "_NET_SYSTEM_TRAY_OPCODE", False);
-    g->xembed_message = XInternAtom(g->display, "_XEMBED", False);
-    g->xembed_info = XInternAtom(g->display, "_XEMBED_INFO", False);
-    g->wm_state = XInternAtom(g->display, "_NET_WM_STATE", False);
-    g->wm_state_fullscreen = XInternAtom(g->display, "_NET_WM_STATE_FULLSCREEN", False);
-    g->wm_state_demands_attention = XInternAtom(g->display, "_NET_WM_STATE_DEMANDS_ATTENTION", False);
-    g->wm_state_hidden = XInternAtom(g->display, "_NET_WM_STATE_HIDDEN", False);
-    g->frame_extents = XInternAtom(g->display, "_NET_FRAME_EXTENTS", False);
-    g->wm_state_maximized_vert = XInternAtom(g->display, "_NET_WM_STATE_MAXIMIZED_VERT", False);
-    g->wm_state_maximized_horz = XInternAtom(g->display, "_NET_WM_STATE_MAXIMIZED_HORZ", False);
+    g->tray_selection = intern_atom(g, tray_sel_atom_name, False);
+    g->tray_opcode = intern_atom(g, "_NET_SYSTEM_TRAY_OPCODE", False);
+    g->xembed_message = intern_atom(g, "_XEMBED", False);
+    g->xembed_info = intern_atom(g, "_XEMBED_INFO", False);
+    g->wm_state = intern_atom(g, "_NET_WM_STATE", False);
+    g->wm_state_fullscreen = intern_atom(g, "_NET_WM_STATE_FULLSCREEN", False);
+    g->wm_state_demands_attention = intern_atom(g, "_NET_WM_STATE_DEMANDS_ATTENTION", False);
+    g->wm_state_hidden = intern_atom(g, "_NET_WM_STATE_HIDDEN", False);
+    g->wm_workarea = intern_atom(g, "_NET_WORKAREA", True);
```

`True` here makes it fail hard if `_NET_WORKAREA` is not supported by the WM.

And BTW we do have such case - with hybrid GUI domain, where gui-daemon in dom0 for sys-gui is launched without window manager at all, and from dom0's Xorg point of view, there is just one fullscreen window.